vmdb/app/models/ems_kubernetes.rb

```diff
@@ -35,27 +42,33 @@ def api_endpoint
     self.class.raw_api_endpoint(hostname, port)
   end
 
-  def connect(_options = {})
-    self.class.raw_connect(hostname, port)
-  end
+  def connect(options = {})
+    hostname = options[:hostname] || self.address
+    port     = options[:port]     || self.port
 
-  def self.event_monitor_class
-    MiqEventCatcherKubernetes
+    self.class.raw_connect(hostname, port)
   end
 
-  def authentication_check
+  def verify_credentials(auth_type = nil, options = {})
     # TODO: support real authentication using certificates
-    [true, ""]
-  end
+    options = options.merge(:auth_type => auth_type)
 
-  def verify_credentials(_auth_type = nil, _options = {})
-    # TODO: support real authentication using certificates
-    true
+    with_provider_connection(options, &:api_valid?)
+  rescue SocketError,
+         Errno::ECONNREFUSED,
+         RestClient::ResourceNotFound,
+         RestClient::InternalServerError => err
+    raise MiqException::MiqUnreachableError, err.message, err.backtrace
+  rescue RestClient::Unauthorized => err
+    raise MiqException::MiqInvalidCredentialsError, err.message, err.backtrace
   end
 
-  def authentication_status_ok?(_type = nil)
+  private
```

Comment:
Adding private here you mask other public methods (all_computer_system_ids, aggregate_logical_cpus, aggregate_memory).